server/stream.go

```diff
@@ -970,7 +970,40 @@ func (jsa *jsAccount) configUpdateCheck(old, new *StreamConfig) (*StreamConfig,
 	}
 
 	// Check limits.
-	if err := jsa.checkAllLimits(&cfg); err != nil {
+	if old.MaxBytes == cfg.MaxBytes {
+		// If MaxBytes isn't being changed, then we can run the normal limit
+		// checking code.
+		if err := jsa.checkAllLimits(&cfg); err != nil {
+			return nil, err
+		}
+		return &cfg, nil
+	}
```

Comment:
is this section really needed? wouldn't it basically be the same issue, just for nothing has changed?
no adjustment has been made, where I'd expect a value of 0.

If I understand this right, the code below this should produce just that